src/java.base/share/classes/jdk/internal/loader/NativeLibraries.java

```diff
@@ -334,6 +240,178 @@ private NativeLibrary findFromPaths(String[] paths, Class<?> fromClass, String n
         return null;
     }
 
+    static class JniNativeLibraries extends NativeLibraries {
+        // All JNI native libraries we've loaded.
+        private static final Set<String> loadedLibraryNames = ConcurrentHashMap.newKeySet();
+
+        private JniNativeLibraries(ClassLoader loader) {
+            // for null loader, default the caller to this class and
+            // do not search java.library.path
+            super(loader, loader != null ? null : NativeLibraries.class,
+                    loader != null ? true : false);
```

Comment:
The last argument expression of the super class could be just loader != null

Reply:
Updated.  Thanks.